sklearn/ensemble/voting.py

```diff
@@ -30,7 +30,13 @@
 def _parallel_fit_estimator(estimator, X, y, sample_weight=None):
     """Private function used to fit an estimator within a job."""
     if sample_weight is not None:
-        estimator.fit(X, y, sample_weight=sample_weight)
+        try:
+            estimator.fit(X, y, sample_weight=sample_weight)
+        except TypeError:
+            raise ValueError(
+                "Underlying estimator {} does not support sample weights."
```

Comment:
It could be some other reason. You are masking the error. You should either:
* say it "may not" support sample weights, and use `raise from`
* check that the right error is being caught